Eject SDCARD => Crash in Tips&Tricks Activity

By GPlay User Report:

### Reproducer
- Eject SDCARD
- Open Tips&Tricks

### Version Information
    App Version: 1.0.0.4
    Syncthing Version: v1.0.0
    Android Version: Android 8.0 / Samsung Galaxy S8 (dreamlte)

### Android Log
2019-01-20 12:25:10.379 31843-31843/? E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.github.catfriend1.syncthingandroid.debug, PID: 31843
    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.github.catfriend1.syncthingandroid.debug/com.nutomic.syncthingandroid.activities.TipsAndTricksActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String java.io.File.getAbsolutePath()' on a null object reference
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2913)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3048)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1808)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6669)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String java.io.File.getAbsolutePath()' on a null object reference
        at com.nutomic.syncthingandroid.activities.TipsAndTricksActivity.onCreate(TipsAndTricksActivity.java:52)
        at android.app.Activity.performCreate(Activity.java:7136)
        at android.app.Activity.performCreate(Activity.java:7127)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1271)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2893)

--- a/app/src/main/java/com/nutomic/syncthingandroid/activities/FolderPickerActivity.java
+++ b/app/src/main/java/com/nutomic/syncthingandroid/activities/FolderPickerActivity.java
@@ -108,6 +108,7 @@ public class FolderPickerActivity extends SyncthingActivity
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
             roots.addAll(Arrays.asList(getExternalFilesDirs(null)));
             roots.remove(getExternalFilesDir(null));
+            roots.remove(null);      // getExternalFilesDirs may return null for an ejected SDcard.
         }
 
         String rootDir = getIntent().getStringExtra(EXTRA_ROOT_DIRECTORY);

--- a/app/src/main/java/com/nutomic/syncthingandroid/activities/TipsAndTricksActivity.java
+++ b/app/src/main/java/com/nutomic/syncthingandroid/activities/TipsAndTricksActivity.java
@@ -48,6 +48,7 @@ public class TipsAndTricksActivity extends SyncthingActivity {
             ArrayList<File> externalFilesDir = new ArrayList<>();
             externalFilesDir.addAll(Arrays.asList(getExternalFilesDirs(null)));
             externalFilesDir.remove(getExternalFilesDir(null));
+            externalFilesDir.remove(null);      // getExternalFilesDirs may return null for an ejected SDcard.
             if (externalFilesDir.size() > 0) {
                 String absExternalStorageAppFilesPath = externalFilesDir.get(0).getAbsolutePath();
                 mTipListAdapter.add(getString(R.string.tip_write_to_sdcard_title),

--- a/app/src/main/java/com/nutomic/syncthingandroid/util/FileUtils.java
+++ b/app/src/main/java/com/nutomic/syncthingandroid/util/FileUtils.java
@@ -167,6 +167,7 @@ public class FileUtils {
             ArrayList<File> externalFilesDir = new ArrayList<>();
             externalFilesDir.addAll(Arrays.asList(context.getExternalFilesDirs(null)));
             externalFilesDir.remove(context.getExternalFilesDir(null));
+            externalFilesDir.remove(null);      // getExternalFilesDirs may return null for an ejected SDcard.
             if (externalFilesDir.size() == 0) {
                 Log.w(TAG, "Could not determine app's private files directory on external storage.");
                 return null;

